rtos/Semaphore.h

```diff
@@ -133,6 +133,14 @@ class Semaphore : private mbed::NonCopyable<Semaphore> {
     */
     osStatus release(void);
 
+    /** Get the number of Semaphore tokens available.
+      @return Number of available tokens.
+
+      @note This function returns the number of available tokens on best effort.
+            This token value is only provided as a reference and is not always accurate.
```

Comment:
Having just approved, minor nit - this doesn't have any ISR section any more. You should have the explicit `@note You may call this function from ISR context.` if that's what you intend.